### DISCUSSION TITLE
[Script Editor] Pressing `ctrl + d` should create additional carets at all occurences of the selected text

### DISCUSSION THREAD

**Discussion by BeayemX** [godotengine/godot-proposals#5582] [Editor]
**Multi cursor / caret**
Godot 4 recently received the feature to use multiple cursors / carets in the script editor.

In my opinion it is not immediately obvious how to use this feature though.
For now I think the only way to get multiple cursor is by holding `Alt` while left clicking with the mouse.

**Requested feature**
In VSCode it is possible to create multiple carets by using `ctrl + d`.

* The first time the shortcut is pressed it will select the whole word under the caret. (Same behavior as in Godot right now)
  * If there is already a selection (e.g. only three letters of a word) the first step will be skipped.
* Consecutive key presses of the shortcut will start selecting the next occurrence of the selected text until all occurrences are selected

**Current behavior**
The behavior in Godot right now is that a second time pressing `ctrl+d` will deselect the current word.
I am not sure if this is necessary because the same result can be achieved by pressing the right arrow key.

**Changes / Regression**
Besides the deselecting part, the current behavior in Godot would not change, but only receive a useful addition.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot/pull/67139.

I think this is a good thing to implement, but it should be done in a separate pull request from https://github.com/godotengine/godot/pull/67139. For reference, <kbd>Ctrl + D</kbd> was implemented in https://github.com/godotengine/godot/pull/48739 (before multi-caret support was added).

**Reply by seppoday** [reply]
I saw that someone on the twitter was trying to implement that 🙂

**Comment by Calinou** [comment]
This is now being tracked in https://github.com/godotengine/godot-proposals/issues/5621.
